contracts/Core/AssetManager.sol

```diff
@@ -90,13 +90,29 @@ contract AssetManager is ACL, AssetStorage, Constants, StateManager {
 
             emit JobActivityStatus(jobs[id].active, id, epoch, block.timestamp);
         } else {
-            collections[id].active = assetStatus;
             if (assetStatus) {
-                numActiveAssets = numActiveAssets + 1;
+                if (!collections[id].active) {
+                    activeAssets.push(id);
+                    collections[id].assetIndex = uint8(activeAssets.length);
+                }
             } else {
-                numActiveAssets = numActiveAssets - 1;
+                if (collections[id].active) {
+                    for (uint8 j = 0; j < activeAssets.length; j++) {
```

Comment:
Looping can be avoided since user can supply the array position as a param

Reply:
Removed looping and fixed with a more optimal solution